pkg/operator/staticpod/startupmonitor/startup_monitor.go

```diff
@@ -0,0 +1,370 @@
+			// TODO: commit suicide ? this seems to be fatal
+			return nil
+		}
+
+		// step 1: create the last known good revision file
+		if err := m.createLastKnowGoodRevisionFor(prevRev, false); err != nil {
+			return err
+		}
+	}
+
+	// step 2: if the last known good revision exits and we got here
+	//         that could mean that:
+	//          - the current revision is broken
+	//          - we just created the last known good revision file
+	//          - the previous iteration of the sync loop returned an error
+	//
+	//         in that case just:
+	//          - annotate the manifest
+	//          - copy the last known good revision manifest
+	lastKnownGoodPod, err := m.readTargetPod(m.lastKnownGoodManifestDstPath())
+	if err != nil {
+		return err
+	}
+	if lastKnownGoodPod.Annotations == nil {
+		lastKnownGoodPod.Annotations = map[string]string{}
+	}
+	lastKnownGoodPod.Annotations["startup-monitor.static-pods.openshift.io/fallback-for-revision"] = fmt.Sprintf("%d", m.revision)
+
+	// the kubelet has a bug that prevents graceful termination from working on static pods with the same name, filename
+	// and uuid.  By setting the pod UID we can work around the kubelet bug and get our graceful termination honored.
+	// Per the node team, this is hard to fix in the kubelet, though it will affect all static pods.
+	lastKnownGoodPod.UID = uuid.NewUUID()
+
+	// remove the existing file to ensure kubelet gets "create" event from inotify watchers
+	rootTargetManifestPath := path.Join(m.manifestsPath, fmt.Sprintf("%s-pod.yaml", m.targetName))
+	if err := m.io.Remove(rootTargetManifestPath); err == nil {
+		klog.Infof("Removed existing static pod manifest %q", path.Join(rootTargetManifestPath))
+	} else if !os.IsNotExist(err) {
+		return err
+	}
+
+	lastKnownGoodPodBytes := []byte(resourceread.WritePodV1OrDie(lastKnownGoodPod))
+	klog.Infof("Writing a static pod manifest %q \n%s", path.Join(rootTargetManifestPath), lastKnownGoodPodBytes)
+	if err := m.io.WriteFile(path.Join(rootTargetManifestPath), lastKnownGoodPodBytes, 0644); err != nil {
+		return err
+	}
+
+	// TODO: commit suicide ?
+	return nil
+}
+
+func (m *monitor) createLastKnowGoodRevisionFor(revision int, strict bool) error {
+	var revisionedTargetManifestPath = m.targetManifestPathFor(revision)
+
+	// step 0: in strict mode remove the previous last good known revision if exists
+	if strict {
+		if exists, err := m.fileExists(m.lastKnownGoodManifestDstPath()); err != nil {
+			return err
+		} else if exists {
+			if err := m.io.Remove(m.lastKnownGoodManifestDstPath()); err != nil {
+				return err
+			}
+			klog.Info("Removed existing last known good revision manifest %s", m.lastKnownGoodManifestDstPath())
+		}
+	}
+
+	// step 1: create last known good revision
+	if err := m.io.Symlink(revisionedTargetManifestPath, m.lastKnownGoodManifestDstPath()); err != nil {
```

Comment:
isn't there a force variant for this? I always thought symlink update is atomic.